homeassistant/components/utility_meter/sensor.py

```diff
@@ -153,11 +157,27 @@ def __init__(
         self._tariff = tariff
         self._tariff_entity = tariff_entity
 
+    def start(self, unit):
+        """Initialize unit and state upon source initial update."""
+        self._unit_of_measurement = unit
+        self._state = 0
+        self.async_write_ha_state()
+
     @callback
     def async_reading(self, event):
         """Handle the sensor state changes."""
         old_state = event.data.get("old_state")
         new_state = event.data.get("new_state")
+
+        if (
+            self._unit_of_measurement is None or self._state is None
```

Comment:
Why check `self._unit_of_measurement`, it's enough to check `self._state`?

Reply:
sanity purposes :)